tests/io/test_pyarrow.py

```diff
@@ -381,10 +382,11 @@ def test_pyarrow_unified_session_properties() -> None:
         **UNIFIED_AWS_SESSION_PROPERTIES,
     }
 
-    with patch("pyarrow.fs.S3FileSystem") as mock_s3fs:
+    with patch("pyarrow.fs.S3FileSystem") as mock_s3fs, patch("pyarrow.fs.resolve_s3_region") as mock_s3_region_resolver:
```

Comment:
ah i had an issue with this before https://github.com/apache/iceberg-python/issues/1279#issuecomment-2466826184

this is a good workaround!